res/controllers/Hercules-DJControl-Inpulse-300-script.js

```diff
@@ -0,0 +1,278 @@
+////////////////////////////////////////////////////////////////////////
+// JSHint configuration                                               //
+////////////////////////////////////////////////////////////////////////
+/* global engine                                                      */
+/* global script                                                      */
+/* global print                                                       */
+/* global midi                                                        */
+//////////////////////////////////////////////////////////////////////// 
+
+var DJCi300 = {};
+///////////////////////////////////////////////////////////////
+//                       USER OPTIONS                        //
+///////////////////////////////////////////////////////////////
+
+// How fast scratching is.
+DJCi300.scratchScale = 1.0;
+
+// How much faster seeking (shift+scratch) is than scratching.
+DJCi300.scratchShiftMultiplier = 4;
+
+// How fast bending is.
+DJCi300.bendScale = 1.0;
+
+// DJControl_Inpulse_300_script.js
+//
+// ***************************************************************************
+// * Mixxx mapping script file for the Hercules DJControl Inpulse 300.
+// * Author: DJ Phatso, contributions by Kerrick Staley
+// * Version 1.2 (December 2019)
+// * Forum: https://www.mixxx.org/forums/viewtopic.php?f=7&t=12599
+// * Wiki: https://mixxx.org/wiki/doku.php/hercules_djcontrol_inpulse_300
+//
+// Changes to v1.2
+// - Code cleanup.
+//
+// Changes to v1.1
+// - Fix seek-to-start and cue-master behavior.
+// - Tweak scratch, seek, and bend behavior.
+// - Controller knob/slider values are queried on startup, so MIXXX is synced.
+// - Fixed vinyl button behavior the first time it's pressed.
+//
+// v1.0 : Original forum release				  
+//
+// TODO: Functions that could be implemented to the script:
+//
+// * ROLL: Keep SLIP active (if already enabled) when exiting from rolls
+//
+// * SLICER/SLICER LOOP: Potentially use and adapt the script of the Pioneer DDJ SX: 
+//  https://www.mixxx.org/wiki/doku.php/pioneer_ddj-sx#slicer_description
+//
+// * TONEPLAY
+//
+// * FX:
+//	- Potentially use 1 FX rack for FX pads and another for the Controlled FX
+//  - See how to preselect effects for a rack
+// * Fix behavior when adjusting tempo slider after pressing [Sync] (tempo adjustment should be relative, not absolute).	  
+//
+// ****************************************************************************
+//
+// We have to disable the no-unused-vars check because we have many MIDI
+// callbacks that receive a fixed list of arguments, but we usually don't use
+// most of these arguments. Eslint seems to make it relatively difficult to
+// disable this check on a case-by-case basis, so we disable it for the whole
+// file.
+// See this GitHub issue for more context:
+// https://github.com/eslint/eslint/issues/1939
+// *eslint-disable no-unused-vars*/
```

Comment:
Please remove lines 50-67. We allow unused parameters if they are starting with an underscore.

Example:
```javascript
// The function below uses value and control, but not group, so we prefix group with an underscore
//DJCi300.vuMeterUpdate = function (value, group, control) {
DJCi300.vuMeterUpdate = function (value, _group, control) {
    value = (value * 127) + 5;
    switch (control) {
        case "VuMeterL":
            midi.sendShortMsg(0xB0, 0x40, value);
            break;
        case "VuMeterR":
            midi.sendShortMsg(0xB0, 0x41, value);
            break;
    }
};
```